astropy/time/core.py

```diff
@@ -1787,6 +1787,85 @@ def light_travel_time(self, skycoord, kind='barycentric', location=None, ephemer
         tcor_val = (spos * cpos).sum(axis=-1) / const.c
         return TimeDelta(tcor_val, scale='tdb')
 
+    def to_local(self, theta, longitude):
+        """
+        Adds longitude including the TIO locator rigorously corrected
+        for polar motion.
+
+        Parameters
+        ---------------
+        theta : float or float array
+            Earth rotation angle or Greenwich sidereal time in radians.
+        longitude : `~astropy.coordinates.Longitude`
+            The longitude on the Earth at which to compute the result.
+
+        Returns
+        -------
+        local_angle : float or float array
+            Either local Earth rotation angle or local sidereal time depending
+            on theta.
+        """
+
+        from astropy.coordinates.builtin_frames.utils import get_polar_motion
+        from astropy.coordinates.matrix_utilities import rotation_matrix
+
+        jd1_tt = getattr(self, 'tt')._time.jd1
```

Comment:
This piece of code is very similar to https://github.com/astropy/astropy/blob/bb4c1973faffea88edc9068df6e95d4452e82928/astropy/coordinates/erfa_astrom.py#L114-L140

This suggests it may be possible to refactor them together.

But probably best to first just get everything to work properly.

Reply:
In fact, it's a copy and paste job. We could just move this to `builtin_frames.utils` along with `get_polar_motion()`, but, like you said, that can happen later.